pkg/installmanager/installmanager.go

```diff
@@ -1810,3 +1820,62 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
 
 	return nil
 }
+
+func (m *InstallManager) patchAzureOverrideCreds() error {
```

Comment:
Unit tests to ensure patch is happening as expected would be a good idea

Reply:
Yes, good idea, let me check if I can add this in installmanager_test.go

Reply:
Added Test_patchAzureOverrideCreds in installmanager_test.go